source/common/http/header_map_impl.h

```diff
@@ -23,21 +23,25 @@ namespace Http {
 #define DEFINE_INLINE_HEADER_FUNCS(name)                                                           \
 public:                                                                                            \
   const HeaderEntry* name() const override { return getInline(HeaderHandles::get().name); }        \
+  size_t remove##name() override { return removeInline(HeaderHandles::get().name); }               \
+  absl::string_view get##name##Value() const override {                                            \
+    return getInlineValue(HeaderHandles::get().name);                                              \
+  }                                                                                                \
+  void set##name(absl::string_view value) override { setInline(HeaderHandles::get().name, value); }
+
+#define DEFINE_INLINE_HEADER_STRING_FUNCS(name)                                                    \
+  DEFINE_INLINE_HEADER_FUNCS(name)                                                                 \
   void append##name(absl::string_view data, absl::string_view delimiter) override {                \
     appendInline(HeaderHandles::get().name, data, delimiter);                                      \
   }                                                                                                \
   void setReference##name(absl::string_view value) override {                                      \
     setReferenceInline(HeaderHandles::get().name, value);                                          \
-  }                                                                                                \
-  void set##name(absl::string_view value) override {                                               \
-    setInline(HeaderHandles::get().name, value);                                                   \
-  }                                                                                                \
-  void set##name(uint64_t value) override { setInline(HeaderHandles::get().name, value); }         \
-  size_t remove##name() override { return removeInline(HeaderHandles::get().name); }               \
-  absl::string_view get##name##Value() const override {                                            \
-    return getInlineValue(HeaderHandles::get().name);                                              \
   }
 
+#define DEFINE_INLINE_HEADER_NUMERIC_FUNCS(name)                                                   \
+  DEFINE_INLINE_HEADER_FUNCS(name)                                                                 \
+  void set##name(uint64_t value) override { setInline(HeaderHandles::get().name, value); }
```

Comment:
There is no integer getter. Should there be?

Reply:
There never was before -- an integer getter would help in the case I mentioned in the description, but I think it strictly is worse.
```response_headers>setStatus(other->getStatusValue());```

Simplifying it to 
``````response_headers>setStatus(other->getStatusInt());```
would mean there'd be a (string -> int -> string) conversion, compared to the no conversion above.